### PR TITLE
Implement feature #9: menu services connection & export settlements csv

### PR DESCRIPTION
**Pull request by Jeffrey-Hong1005**
This PR resolves Issue #9.

- Implemented Exporter.writePaymentsCsv() to generate CSV files with header "from,to,amount".

- Added ExporterTest.java covering both normal and edge cases:
   -> Happy path: Multiple Payment objects correctly exported to CSV (rows include 'from,to,amount').
   -> Edge case: Empty Payment list still creates a file containing only the header line.

- Verified App.java integration: Export option (menu 7) correctly calls Exporter and creates data/settlement.csv.
- Other menu options (List, Summary, Settle) depend on pending issues, so 'load() not implemented yet' messages are expected.

How to Test
1. Run `mvn test` → confirms all JUnit tests pass (ExporterTest.java).  
2. Run `mvn compile exec:java -Dexec.mainClass=itm.oss.splitter.App`  
   → menu, select option 7 to manually verify CSV export (data/settlement.csv created).

Result
- Code compiles and all tests pass (build success).
- Manual run prints menu correctly.
- Export feature produces a valid CSV file with header and rows.

### CODE REVIEW

**Comment by hayat711** [comment]
**Group3** will be reviewing this PR [code implemented by group4 for feature #9 ]

File: src/test/java/itm/oss/splitter/ExporterTest.java
Comment (GeonKim0422) [inline src/test/java/itm/oss/splitter/ExporterTest.java:1]: It seems like there are not any comments about ExporterTest.java file.

File: src/main/java/itm/oss/splitter/Exporter.java [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
 public class Exporter {
 
   public static void writePaymentsCsv(String path, ArrayList<Payment> pays) throws IOException {
     // TODO (Issue 9): write CSV with header: from,to,amount
-    throw new UnsupportedOperationException("writePaymentsCsv() not implemented yet");
+
+    File file = new File(path);
+
+    // Automatically close using try-with-resources
+    try (FileWriter writer = new FileWriter(file)) {
```
Comment (MelonChicken) [inline src/main/java/itm/oss/splitter/Exporter.java:19]: `try (FileWriter writer = new FileWriter(file))` will fail if the parent directory of path does not exist. `FileWriter` does not create directories and will throw an exception (e.g., `FileNotFoundException`). I also couldn’t find any test or verification that covers this case.

The below is the error that I faced when I run the test after deleting `data/` repository.
image 

```suggestion
    +// Ensure parent directory exists to avoid FileNotFoundException
+File parent = file.getParentFile();
+if (parent != null && !parent.exists()) { // Check if there is no parent direcotry
+    java.nio.file.Files.createDirectories(parent.toPath());
+}

    try (FileWriter writer = new FileWriter(file)) {
```

It might be safer to check and create (if necessary) the parent directory before opening the writer so that the export works consistently across environments. :+1:

File: src/main/java/itm/oss/splitter/Exporter.java [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+
+      // CSV header
+      writer.write("from,to,amount");
```
Comment (MelonChicken) [inline src/main/java/itm/oss/splitter/Exporter.java]: image


As we can see in the exported result (expense.sample.csv), the header is currently written without a newline.

This might cause **concatenation problems** since many CSV parsers (e.g., `OpenCSV`) expect the header to end **with a line break**. Without it, the first data row can be concatenated to the header in some readers.

Consider writing a newline after the header for better CSV compatibility.

```suggestion
      writer.write("from,to,amount\n"); //add a newline after the header for better CSV compatibility.
```

**Review by MelonChicken — CHANGES_REQUESTED** [review]
## Two small issues found during export:

### Header newline
The exported file currently writes the header without a `newline`, which can cause **concatenation issues** in some CSV readers (e.g., `OpenCSV`), where the first data row may be appended to the header.

### Parent directory handling
`try (FileWriter writer = new FileWriter(file))` will fail if the **parent directory does not exist**. `FileWriter` does not create directories and will throw (e.g., `FileNotFoundException`). I couldn’t find a test covering this case.

File: src/test/java/itm/oss/splitter/ExporterTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Test
+    public void testWritePaymentsCsv_handlesEmptyList() throws Exception {
+        
+        ArrayList<Payment> pays = new ArrayList<>();
+        String path = "data/test_empty.csv";
+
+        Exporter.writePaymentsCsv(path, pays);
+
+        File file = new File(path);
```
Comment (GeonKim0422) [inline src/test/java/itm/oss/splitter/ExporterTest.java]: I recommend replacing the intance name 'file' to 'createdFile' to raise readibility.
```suggestion
        File createdfile = new File(path);
```

File: src/test/java/itm/oss/splitter/ExporterTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.File;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import static org.junit.jupiter.api.Assertions.*;
+import java.math.*;
+
+public class ExporterTest {
+
+    @Test
+    public void testWritePaymentsCsv_createsFileWithHeader() throws Exception {
```
Comment (GeonKim0422) [inline src/test/java/itm/oss/splitter/ExporterTest.java]: I recommend replacing this method name to writePaymentsCsvCreatesFileWithHeader.

1. 'Test' can be deleted. Annotation '@Test' implies that this method is for the test.
2.  Camel case is standard rather than underscore in Java.
```suggestion
    public void writePaymentsCsvCreatesFileWithHeader throws Exception {
```

**Review by GeonKim0422 — CHANGES_REQUESTED** [review]
I have focused on readibility of your code. Please check my comment.

File: src/test/java/itm/oss/splitter/ExporterTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        assertTrue(file.exists(), "csv file needs to be created");
+
+        String content = Files.readString(file.toPath());
+        assertTrue(content.contains("from,to,amount"), "needs to include csv header");
+        assertTrue(content.contains("Alice,Bob,30"), "needs to include payment record");
+    }
+
+    @Test
+    public void testWritePaymentsCsv_handlesEmptyList() throws Exception {
+
```
Comment (bigmansinwoo-coder) [inline src/test/java/itm/oss/splitter/ExporterTest.java:38]: To further strengthen the reliability and business correctness of this feature, how about adding one
more test that explicitly covers zero and negative amounts?

Examples of cases worth verifying:
- Negative amount → whether it is allowed and how it is represented in the CSV. 

```suggestion
@Test
void writePaymentsCsv_HandlesNegativeAmounts() throws Exception {
    Path csvPath = Paths.get("data", "edgecases", "negative.csv");
    Files.createDirectories(csvPath.getParent());
    Files.deleteIfExists(csvPath);

    List<Payment> payments = List.of(
        new Payment("Bob", "Charlie", new BigDecimal("-15.5"))
    );

    Exporter.writePaymentsCsv(csvPath.toString(), payments);

    List<String> lines = Files.readAllLines(csvPath);
    assertEquals(2, lines.size()); // header + 1 row
    assertEquals("from,to,amount", lines.get(0));
    assertEquals("Bob,Charlie,-15.50", lines.get(1)); // expecting two decimal places
}


```

File: src/main/java/itm/oss/splitter/Exporter.java [hunk 6]
```diff
@@ -1,12 +1,40 @@
...
+      // Write data only when the list is not null or empty
+      // if value is null, write empty string
+        for (int i = 0; i < pays.size(); i++) {
+          Payment p = pays.get(i);
+          String from = (p.getFrom() == null) ? "" : p.getFrom();
+          String to = (p.getTo() == null) ? "" : p.getTo();
+          String amount = "";
+
+          if(p.getAmount() != null) {
+            amount = p.getAmount().setScale(2, RoundingMode.HALF_EVEN).toString();
```
Comment (bigmansinwoo-coder) [inline src/main/java/itm/oss/splitter/Exporter.java:38]: A short comment explaining **why** we use 2-decimal precision and **why** HALF_EVEN is chosen
(e.g., accounting standard / CSV compatibility / business rule) would prevent confusion and
accidental format changes in future revisions.

```suggestion
            // Enforce banking-style rounding (HALF_EVEN) with 2 decimal places
// to ensure financial accuracy and CSV compatibility
```

File: src/test/java/itm/oss/splitter/ExporterTest.java
Comment (Jeffrey-Hong1005) [reply src/test/java/itm/oss/splitter/ExporterTest.java:1]: Thank you for your recheck!! :)

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 1 above)
Comment (Jeffrey-Hong1005) [reply src/main/java/itm/oss/splitter/Exporter.java:19]: Applied your suggestion. We added directory existence check before opening FileWriter. Verified that export now works even if data/ folder is missing. Thanks for the catch!

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 2 above)
Comment (Jeffrey-Hong1005) [reply src/main/java/itm/oss/splitter/Exporter.java]: Added \n after the CSV header as suggested. Without it, the first data row appeared directly after the header. Now, exported file correctly starts data on a new line.

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 5 above)
Comment (Jeffrey-Hong1005) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: Applied negative case code, creating edgecases file in data folder. Thank you for your review!!

**Review by Jeffrey-Hong1005 — COMMENTED** [review]
Applied all of the comments in our PR.

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 1 above)
Comment (MelonChicken) [reply src/main/java/itm/oss/splitter/Exporter.java:19]: LGTM, thanks for adding comments, now we can see the process clearly!

**Review by MelonChicken — APPROVED** [review]
LGTM. Thanks to your kind response, now we have safer exporter 👍

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 2 above)
Comment (MelonChicken) [reply src/main/java/itm/oss/splitter/Exporter.java]: LGTM. Thank you for your kind response!

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 6 above)
Comment (bigmansinwoo-coder) [reply src/main/java/itm/oss/splitter/Exporter.java]: LGTM, code become more easy to read.

File: src/test/java/itm/oss/splitter/ExporterTest.java
Comment (GeonKim0422) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: LGTM! Thanks for the kind comments.

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 3 above)
Comment (yeonlimee2) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: Thanks for the suggestion!
I've updated the variable name from file to createdFile for better readability.
The change has been pushed to the branch.

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 4 above)
Comment (yeonlimee2) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: Thanks for your feedback!
I’ve renamed the test method to removed the redundant ‘test’.
The updated change has been pushed to the branch.

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 3 above)
Comment (GeonKim0422) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: I found your code refined as I intended. Look goods to me!

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 4 above)
Comment (GeonKim0422) [reply src/test/java/itm/oss/splitter/ExporterTest.java]: LGTM! Thanks for the fast reply. I recognized that you have changed the function name that I missed. Now your code is easier to read.

**Review by GeonKim0422 — APPROVED** [review]
LGTM!

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 6 above)
Comment (ERE252) [reply src/main/java/itm/oss/splitter/Exporter.java:38]: Added comments explaining the reason for using 2-decimal precision and HALF_EVEN rounding as suggested.
Thank you for the helpful feedback!

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 5 above)
Comment (ERE252) [reply src/test/java/itm/oss/splitter/ExporterTest.java:38]: Added a new test method to verify handling of negative amounts as suggested.
Thanks for the helpful review!

File: src/test/java/itm/oss/splitter/ExporterTest.java
Comment (yeonlimee2) [reply src/test/java/itm/oss/splitter/ExporterTest.java:1]: I found parts of the code with the same structure as the sections I received feedback on, but they were not modified yet, so I fixed them. Thank you for your feedbacks!

**Review by bigmansinwoo-coder — APPROVED** [review]

File: src/test/java/itm/oss/splitter/ExporterTest.java (hunk 5 above)
Comment (bigmansinwoo-coder) [reply src/test/java/itm/oss/splitter/ExporterTest.java:38]: Great, thanks for the quick update. The new edge cases make the test coverage more solid.

File: src/main/java/itm/oss/splitter/Exporter.java (hunk 6 above)
Comment (bigmansinwoo-coder) [reply src/main/java/itm/oss/splitter/Exporter.java:38]: LGTM, code become more easy to read.
